src/components/error-boundary/error-boundary.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { ReactNode, ErrorInfo } from 'react';
+import { formatMessage } from '../../utility/errors';
+import { AlertDialog, AlertDialogProps } from '../alert-dialog/alert-dialog';
+
+
+export interface AlertErrorProps {
+  message?: string,
+  alertTitle?: string,
+  error?: Error,
+  info?: ErrorInfo,
+}
+
+export class AlertError extends Error {
+  readonly alertTitle?: string;
+  readonly error?: Error;
```

Comment:
This seems a little odd to have an Error type that also has an `error` member. Maybe this should follow the pattern for deriving errors used in `/server/util/error-types.ts`.

Reply:
I like it! Before I get too far into this, I'm thinking something along these lines:

```tsx
export class AlertError extends Error {
  constructor(
    message: string,
    readonly type: AlertErrorType,
    readonly title?: string,
    readonly isException: boolean = false,
  ) {
    super(message);
  }
}

export class RosterColumnError extends AlertError {
  constructor() {
    super('Failed to get roster column info', 'RosterColumn', 'Get Roster Column Info');
    Error.captureStackTrace(this, RosterColumnError);
  }
}

export class RosterUploadError extends AlertError {
  constructor(message: string) {
    super(`${message}. Please verify the roster data.`, 'RosterUpload', 'Upload Error');
    Error.captureStackTrace(this, RosterUploadError);
  }
}

export type AlertErrorType = (
  'RosterColumn' |
  'RosterUpload'
);
```

Then, in the roster-page.tsx file, we'd have something like this:

```tsx
  const initializeRosterColumnInfo = useCallback(async () => {
    try {
      const infos = (await axios.get(`api/roster/${orgId}/info`)).data as ApiRosterColumnInfo[];
      const unitColumnWithInfos = [unitColumn, ...infos];
      setRosterColumnInfos(unitColumnWithInfos);
      if (initialLoad.current) {
        setVisibleColumns(unitColumnWithInfos.slice(0, maxNumColumnsToShow));
      }
    } catch (error) {
      handleError(new RosterColumnError());
    }
  }, [orgId, handleError]);
```

Also, note that the `isException` property can be used to distinguish between expected errors and unexpected exceptions.

Reply:
Something just occurred to me... rather than specifying an `isException` bool, can we use `if (error instanceof AlertError) {...}` to determine whether it should show an alert or be treated as an unhandled exception? I was thinking a bool originally as well, but I think it would be even better if the type could be used. That's similar to how the server is working, where regular Error objects are treated as internal server errors.

I'd maybe opt for not getting too specific with the error object types unless the same one is used lots of times. But it could be a nice pattern to establish for when that is the case.

Reply:
Definitely a cleaner, simpler approach. Fixed in the latest commit.